examples/dataflow-custom-templates/cloud_shell_tutorial.md

```diff
@@ -0,0 +1,223 @@
+
+The word count example requires a source [Google Cloud Storage](https://cloud.google.com/storage) bucket.
+To make the example interesting, we copy all the files from
+`gs://apache-beam-samples/shakespeare/*` to a custom bucket in our project.
+
+Therefore, this step will:
+- Provision a Google Cloud storage bucket
+- Create Google Cloud storage objects to read from in the pipeline
+
+Run the terraform workflow in
+the [infrastructure/03.io](infrastructure/03.io) directory.
+
+Terraform will ask your permission before provisioning resources.
+If you agree with terraform provisioning resources,
+type `yes` to proceed.
+
+```sh
+DIR=infrastructure/03.io
+terraform -chdir=$DIR init
+terraform -chdir=$DIR apply -var='project=<walkthrough-project-id/>'
+```
+
+## 6. Provision the Dataflow template builder
+
+We will use [Cloud Build](https://cloud.google.com/build) to build the
+custom Dataflow template.  There are advantages to using Cloud Build to build
+our custom Dataflow template, instead of performing the necessary commands on
+our local machine.  Cloud Build connects to our version control,
+[GitHub](https://GitHub.com) in this example, so that any changes made to
+a specific branch will automatically trigger a new build of our Dataflow
+template.
+
+Therefore, this step will:
+
+- Provision cloud build trigger that will:
+    1. Run the language specific build process i.e. gradle shadowJar, go build, etc.
+    2. Execute the `gcloud dataflow flex-template` command with relevant arguments.
+
+### 6.1. Fork the repository into your own GitHub organization or personal account
+
+In order to benefit from [Cloud Build](https://cloud.google.com/build), the service
+requires we own this repository; it will not work with a any repository, even
+if it is public.
+
+Therefore, complete these steps before proceeding:
+1) [Fork the repository](https://github.com/GoogleCloudPlatform/professional-services/fork)
+2) [Connect forked repository to Cloud Build](https://console.cloud.google.com/cloud-build/triggers/connect)
+
+### 6.2 Execute terraform module to provision Cloud Build trigger
+
+First, set your GitHub organization or username:
+
+```sh
+GITHUB_REPO_OWNER=<change me>
+```
+
+Next, set expected defaults.  (_Note: Normally it makes sense to default terraform
+variables instead of doing this._)
+
+```sh
+GITHUB_REPO_NAME=professional-services
+WORKING_DIR_PREFIX=examples/dataflow-custom-templates
+```
+
+Run the terraform workflow in
+the [infrastructure/04.template](infrastructure/04.template) directory.
+
+Terraform will ask your permission before provisioning resources.
+If you agree with terraform provisioning resources,
+type `yes` to proceed.
+
+```sh
+DIR=infrastructure/04.template
+terraform -chdir=$DIR init
+terraform -chdir=$DIR apply -var="project=$(gcloud config get-value project)" -var="github_repository_owner=$GITHUB_REPO_OWNER" -var="github_repository_name=$GITHUB_REPO_NAME" -var="working_dir_prefix=$WORKING_DIR_PREFIX"
```

Comment:
To test this PR at this step, one would need to run:
terraform -chdir=$DIR apply -var="project=$(gcloud config get-value project)" -var="github_repository_owner=$GITHUB_REPO_OWNER" -var="github_repository_name=$GITHUB_REPO_NAME" -var="working_dir_prefix=$WORKING_DIR_PREFIX" -var="github_repository_branch=dataflow-custom-templates"